### PR TITLE
Add deployment process

### PR DESCRIPTION
**Pull request by JamesDawson**
* Implements an initial deployment process using [InvokeBuild](https://github.com/nightroman/Invoke-Build) for orchestration
* Expands the supported authentication mechanisms
* Initial integration with the `pac` CLI

### CODE REVIEW

File: module/functions/Connect-DataverseEnvironment.ps1 [hunk 1]
```diff
@@ -46,25 +75,38 @@ function Connect-DataverseEnvironment
...
-        ClientId = $ClientId
-        TenantId = $TenantId
-        # AadAuthorityAudience = [Microsoft.Identity.Client.AadAuthorityAudience]::AzureAdMultipleOrgs
-        RedirectUri = "http://localhost"
+    # Acquire the token for CI/CD scenarios
+    if ($env:AZURE_CLIENT_ID -and $env:AZURE_CLIENT_SECRET -and $env:AZURE_TENANT_ID)
+    {
+        Write-Host "Attemping authentication via environment variables [ClientId=$env:AZURE_CLIENT_ID]"
+        $authResult = Get-MsalToken -ClientId $env:AZURE_CLIENT_ID `
+                                    -Scopes $authScope `
```
Comment (edfreeman) [inline module/functions/Connect-DataverseEnvironment.ps1:83]: Where does `$authScope` come from?

Comment (edfreeman) [reply module/functions/Connect-DataverseEnvironment.ps1:83]: Looks like the corresponding code has been deleted?

File: module/functions/Connect-PowerPlatformCli.ps1 [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+                                Where-Object { $_.Name -eq $ProfileName }
+        $existingEnvironmentProfile = $pacAuthList |
+                                Where-Object { $_.Url -eq $safeEnvironmentUrl }
+
+        if ($requiredProfile) {
+            Write-Host "Found matching profile"
+            $pacAuthProfileIndex = $requiredProfile.Index
+        }
+        elseif (!$existingProfile -and $existingEnvironmentProfile) {
+            throw "A PowerApp CLI profile already exists for the target environment. [Profile=$ProfileName] [Environment=$EnvironmentUrl]"
```
Comment (edfreeman) [inline module/functions/Connect-PowerPlatformCli.ps1:92]: Should we add a message about what action to take for these two exceptions?

File: module/functions/ConvertFrom-Fwsv.ps1 [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# <copyright file="ConvertFrom-Fwsv.ps1" company="Endjin Limited">
+# Copyright (c) Endjin Limited. All rights reserved.
+# </copyright>
+
+<#
+.SYNOPSIS
+Obtains an access token for the Dataverse environment.
+
+.DESCRIPTION
+Obtains an access token for the Dataverse environment that is used for other operations.
```
Comment (edfreeman) [inline module/functions/ConvertFrom-Fwsv.ps1:10]: Not the correct synopsis/description

File: module/functions/Select-PowerAppsSolution.ps1 [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# <copyright file="Select-PowerAppsSolution.ps1" company="Endjin Limited">
+# Copyright (c) Endjin Limited. All rights reserved.
+# </copyright>
+
+<#
+.SYNOPSIS
+Obtains an access token for the Dataverse environment.
+
+.DESCRIPTION
+Obtains an access token for the Dataverse environment that is used for other operations.
```
Comment (edfreeman) [inline module/functions/Select-PowerAppsSolution.ps1:10]: Same here

File: module/tasks/tasks.ps1 [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+        $tableId = Set-DataverseTable @tableDefinition -Verbose:$VerboseLogging
+
+        foreach ($column in $tableDefinition.columns) {
+            $columnId = $tableId | Set-DataverseColumn @column -Verbose:$VerboseLogging
+        }
+    }
+}
+
+task Deploy EnsureDataverseEnvironment,
```
Comment (edfreeman) [inline module/tasks/tasks.ps1:76]: Is this not the same as `process.ps1`? What's the point having that other script?

File: module/functions/Connect-DataverseEnvironment.ps1 (hunk 1 above)
Comment (JamesDawson) [reply module/functions/Connect-DataverseEnvironment.ps1:83]: Good catch, it got renamed.

File: module/tasks/tasks.ps1 (hunk 5 above)
Comment (JamesDawson) [reply module/tasks/tasks.ps1:76]: Just following the pattern we use elsewhere, of separating the process from the task implementations - although you could reasonably argue it's overkill given the simple process we currently have.

**Review by edfreeman — APPROVED** [review]

**Pull request merged**